### DISCUSSION TITLE
Onboarding UI and App Install Progress Bar

### DISCUSSION THREAD

**Discussion by Bipsibopsi** [IBM/mac-ibm-notifications#182] [Q&A] [answered]
Just curious if this is the intended, or possible, way we could use the Onboarding UI.

We would like to add in a page that contains a progress bar that will reflect the progress of the applications installing in the background. I was able to get this to work with the Popup UI, but would really like to combine it into the Onboarding UI / Process if possible.

Any thoughts?

We have tried to modify the payload and arguments in the Onboarding UI, but I feel like there is a limitation for the progress bar in that it will only track the progress of the Onboarding UI. Long story short, can we basically take what we had working in the Popup window and modify/fit it into the Onboarding UI?

**Comment by SMartorelli** [comment, marked as answer]
Hi @Bipsibopsi,

The Onboarding UI has the capability to display a common progress bar shared across all the pages of the onboarding workflow. This feature is also compatible with a single-page Onboarding UI. You can update the progress in the same manner as you currently do with the popup UI.

To define a progress bar in the onboarding payload, you simply need to use the dedicated field in the JSON. You can refer to this [example](https://github.com/IBM/mac-ibm-notifications/wiki/Onboarding-UI-Workflow#example-of-json) for more details.

P.S.: For version 3.0.0, we've received some feedback regarding some issues on the Onboarding UI. Next week, we're releasing version 3.0.2, which will include multiple fixes to enhance its performance and usability.

**Comment by Bipsibopsi** [comment]
Hey @SMartorelli, 

That worked - we have it up and running and it works well. Much appreciated.

**Comment by Anthobdn** [comment]
Hi, I try to do the same thing, with the /percent automatic on my Onboarding, but the progress bar doesn't update at all...
I followed the Script Exemple... Thanks for help.

**Reply by SMartorelli** [reply]
Hi @Anthobdn sorry for the delayed answer. You should use only "automatic" as payload for the "progressBarPayload" json key.
PS: I've noticed that the wiki might be confusing on this so I'll fix it. Thanks!
